files/scripts/allowlist.py

```diff
@@ -15,32 +18,137 @@
 """
 
 
+class RepoUrl(click.types.ParamType):
+    name = "repo url"
+
+    def __verify(self, path: str) -> bool:
+        """
+        Verifies if path is used with intention it has (domain vs namespace vs
+        repository itself).
+
+        Args:
+            path: Repository or namespace URL.
+
+        Returns:
+            `True` if verified, `False` otherwise.
+        """
+        highlighted = "namespace"
+
+        if "/" not in path:
+            highlighted = "whole domain"
+        elif path.endswith(".git"):
+            highlighted = "specific repository"
+
+        return click.confirm(
+            "Are you manipulating with a "
+            + click.style(highlighted, fg="red", bold=True)
+            + f" ‹{path}›?"
+        )
+
+    def convert(
+        self,
+        value: Optional[str],
+        param: Optional[click.Parameter] = None,
+        ctx: Optional[click.Context] = None,
+    ) -> Optional[str]:
+        if not value:
+            return None
+
+        parsed_path = urlparse(value)
+        if parsed_path.scheme:
+            click.secho("Protocol ignored when allowlisting.", fg="gray")
+        result = f"{parsed_path.netloc}{parsed_path.path}"
+
+        if not self.__verify(result):
+            return None
+        return result
+
+
+def construct_path() -> str:
+    """
+    Constructs path from user input.
+
+    Returns:
+        Path from user input.
+    """
+    domain = click.prompt("Please input domain (e.g. github.com, gitlab.com)")
+    namespace_or_repo = click.prompt(
+        "(in case of specific repository, type the whole path suffixed with "
+        "‹.git›, e.g. for ‹ogr› repository in ‹packit› namespace type "
+        "‹packit/ogr.git›)\nPlease input namespace"
+    )
```

Comment:
If this is broken up like this, then I think `PATH_HELP` should be rewritten to avoid confusion.

Reply:
that function is used as fallback, you can do

    $ allowlist.py approve github.com # or
    $ allowlist.py approve # <- with this you get the prompt from that function

Reply:
Okay, I see. No update needed then. :)